src/rendering/Renderer2D.zig

```diff
@@ -684,13 +692,15 @@ pub fn render(self: Self, screen_size: Size) void {
 }
 
 /// Appends a set of triangles to the renderer with the given `texture`.
-pub fn appendTriangles(self: *Self, texture: ?*ResourceManager.Texture, triangles: []const [3]Vertex) DrawError!void {
+pub fn appendTriangles(self: *Self, texture: ?*ResourceManager.Texture, triangles: []const [3]Vertex, rot_radians:?f32, rot_about:?Point) DrawError!void {
     const draw_call = if (self.draw_calls.items.len == 0 or self.draw_calls.items[self.draw_calls.items.len - 1] != .draw_vertices or self.draw_calls.items[self.draw_calls.items.len - 1].draw_vertices.texture != texture) blk: {
```

Comment:
This line is missing checks for rotation. Otherwise, other elements with the same texture, but unrelated drawcalls will render also rotated

Reply:
Change added. Using std.meta.eql() as the tests are on optionals and a struct

Reply:
Can you please use a separate compare function with `std.math.approxEqAbs()` with 0.05 degree for rotation and 0.1 pixels absolute threshold instead? It would make the comparison be a bit more lax, and also more obvious on what it does

Reply:
Done. Although, the pixels components are i16, so I'm checking them exactly.